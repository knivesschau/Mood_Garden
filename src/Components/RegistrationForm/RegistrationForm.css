.LoginForm {
    text-align: center;
    padding: 1em;
}

label {
    display: flex;
    flex-direction: column;
    padding: .5em;
}

button {
    margin-top: 2em;
}
