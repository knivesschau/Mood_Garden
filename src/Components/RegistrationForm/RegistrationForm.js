import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import ErrorValidation from '../../ErrorHandlers/ErrorValidation';
import AuthApiService from '../../services/auth-api-service';
import './RegistrationForm.css';

export default class RegistrationForm extends Component {
    static defaultProps = {
        onValidlRegistration: () => {}
    }
    
    submitRegistration = e => {
        e.preventDefault();

        const {username, password} = e.target; 

        this.setState({
            error: null
        })

        AuthApiService.postUser({
            user_name: username.value,
            password: password.value
        })
            .then(user => {
                username.value = '';
                password.value = '';
                this.props.onValidlRegistration();
            })
            .catch(res => {
                this.setState({
                    error: res.error
                })
            })
    
    }    
    
    render() {
        return (

            <section className="LoginForm">

                <nav role="navigation" className="Register_Nav"> 
                    <Link to='/'>
                        Back
                    </Link>
                </nav>

                <h3>Begin Cultivating</h3>

                <form className="Registration_Form" onSubmit={this.submitRegistration}>

                    <div className="Username">

                        <label htmlFor="username">
                            Username:

                            <input type="text" id="username" name="username"/>

                        </label>

                    </div>

                    <div className="Password">

                        <label htmlFor="password">
                            
                            Password:

                            <input type="password" id="password" name="password"/>
                        
                        </label>

                        <label htmlFor="retype-password">
                            
                            Re-Type Password:

                            <input type="password" id="retype-password" name="re-typepassword"/>
                        
                        </label>

                    </div>                                                 

                <button type="submit">Register</button>

                </form>

            </section>
        )
    }
}