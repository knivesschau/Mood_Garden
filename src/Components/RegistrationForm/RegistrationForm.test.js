import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import RegistrationForm from './RegistrationForm';

describe ('RegistrationForm Component', () => {
    it ('renders the complete RegistrationForm Component', () => {
        const wrapper = shallow(<RegistrationForm/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})