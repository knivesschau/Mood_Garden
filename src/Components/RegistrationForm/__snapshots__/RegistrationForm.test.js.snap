// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm Component renders the complete RegistrationForm Component 1`] = `
<section
  className="LoginForm"
>
  <nav
    className="Register_Nav"
    role="navigation"
  >
    <Link
      to="/"
    >
      Back
    </Link>
  </nav>
  <h3>
    Begin Cultivating
  </h3>
  <form
    className="Registration_Form"
    onSubmit={[Function]}
  >
    <div
      className="Username"
    >
      <label
        htmlFor="username"
      >
        Username:
        <input
          id="username"
          name="username"
          type="text"
        />
      </label>
    </div>
    <div
      className="Password"
    >
      <label
        htmlFor="password"
      >
        Password:
        <input
          id="password"
          name="password"
          type="password"
        />
      </label>
      <label
        htmlFor="retype-password"
      >
        Re-Type Password:
        <input
          id="retype-password"
          name="re-typepassword"
          type="password"
        />
      </label>
    </div>
    <button
      type="submit"
    >
      Register
    </button>
  </form>
</section>
`;
